i6_models/parts/conformer/feedforward.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Callable
+
+import torch
+from torch import nn
+
+from i6_models.config import ModelConfiguration
+
+
+@dataclass
+class ConformerPositionwiseFeedForwardV1Config(ModelConfiguration):
+    input_dim: int
+    """input dimension"""
+    hidden_dim: int
+    """hidden dimension (normally set to 4*input_dim as suggested by the paper)"""
+    dropout: float
+    """dropout probability"""
+    activation: Callable[[torch.Tensor], torch.Tensor]
+    """activation function"""
+
+
+class ConformerPositionwiseFeedForwardV1(nn.Module):
+    """
+    Conformer feedforward module
+    """
+
+    def __init__(self, cfg: ConformerPositionwiseFeedForwardV1Config):
+        super().__init__()
+
+        self.layer_norm = nn.LayerNorm(cfg.input_dim)
+        self.linear_ff = nn.Linear(in_features=cfg.input_dim, out_features=cfg.hidden_dim, bias=True)
+        self.activation = cfg.activation
+        self.linear_out = nn.Linear(in_features=cfg.hidden_dim, out_features=cfg.input_dim, bias=True)
+        self.dropout = nn.Dropout(p=cfg.dropout)
+
+    def forward(self, tensor: torch.Tensor) -> torch.Tensor:
+        """
+        :param tensor: shape [B,T,F], F=input_dim
+        :return: shape [B,T,F], F=input_dim
+        """
+        tensor = self.layer_norm(tensor)
+        tensor = self.linear_ff(tensor)  # [B,T,F]
+        tensor = self.activation(tensor)  # [B,T,F]
+        tensor = self.dropout(tensor)  # [B,T,F]
+        tensor = self.linear_out(tensor)  # [B,T,F]
+        tensor = self.dropout(tensor)  # [B,T,F]
```

Comment:
if the same `self.dropout` is used after `linear_out` and before, will this make a difference as to using two different dropouts or does this not matter? 
(e.g. will always the same dimensions be dropped)

Reply:
Good point. Removed my approval for now since this is quite important (also for the other PRs) to clarify.

Reply:
I have made a small test to verify that the second dropout would randomly mask out channel based on the result of the first dropout (two dropouts work independently)
```
>>> dropout = torch.nn.Dropout(p=0.5)
>>> output = dropout(output)
>>> output = dropout(input)
>>> output
tensor([[0., 2., 0., 0., 0., 2., 0., 2., 2., 0.]])
>>> output = dropout(output)
>>> output
tensor([[0., 0., 0., 0., 0., 0., 0., 4., 4., 0.]])
```

Reply:
Just look at the code ([here](https://pytorch.org/docs/stable/_modules/torch/nn/modules/dropout.html#Dropout)).

But to make the code more explicit, it might anyway be a good idea to use the functional variant instead: [`torch.nn.functional.dropout`](https://pytorch.org/docs/stable/generated/torch.nn.functional.dropout.html)

Reply:
In any case, I would avoid calling the same module here again. That's very uncommon for PyTorch, unless you explicitly intend to have param sharing.

Reply:
> it might anyway be a good idea to use the functional variant instead

Please be aware though that you then need to pass `training` explicitly and Dropout will no longer automatically be only enabled in training.

Reply:
Just `tensor = torch.nn.functional.dropout(tensor, p=self.dropout, training=self.training)` (assuming `self.dropout` is now the float for the dropout prob).